### PR TITLE
ENH: support RGB in pcolormesh (wrapped case)

### PR DESCRIPTION
**Pull request by kylehofmann**
This PR adds to `pcolormesh` support for RGB data (that is not RGBA data).

The handling of masked RGBA data is also slightly different.

### CODE REVIEW

File: lib/cartopy/tests/mpl/test_mpl_integration.py [hunk 1]
```diff
@@ -245,8 +245,13 @@ def test_cursor_values():
     MPL_VERSION.release[:2] < (3, 8), reason='mpl < 3.8')
 PARAMETRIZE_PCOLORMESH_WRAP = pytest.mark.parametrize(
     'as_rgba',
```
Comment (rcomer) [inline lib/cartopy/tests/mpl/test_mpl_integration.py]: Really minor thing, but I would change the 'as_rgba' variable name now that it’s no longer a boolean.  Maybe something like 'data_kind'?  I can always update it after merge though.

**Review by rcomer — COMMENTED** [review]
Thanks @kylehofmann it’s a very pleasant surprise to see this!  The code change looks good to me.  When I have time hopefully later this week I will test it locally against Greg’s Matplotlib branch.  Assuming it all passes I’ll merge it in.

I see you have already signed the SciTools CLA.  Thanks for doing that too 👍

File: lib/cartopy/tests/mpl/test_mpl_integration.py (hunk 1 above)
Comment (kylehofmann) [reply lib/cartopy/tests/mpl/test_mpl_integration.py]: Good idea. I went with "mesh_data_kind".

**Comment by kylehofmann** [comment]
I think this version is ready to merge as-is. Glad I could help!

**Comment by rcomer** [comment]
Thanks @kylehofmann!  Local tests look good too.

**Pull request merged**